apps/files_trashbin/lib/Sabre/TrashbinPlugin.php

```diff
@@ -129,4 +132,40 @@ public function httpGet(RequestInterface $request, ResponseInterface $response):
 			$response->addHeader('Content-Disposition', 'attachment; filename="' . $node->getFilename() . '"');
 		}
 	}
+
+	/**
+	 * Check if a user has available space before attempting to
+	 * restore from trashbin unless they have unlimited quota.
+	 *
+	 * @param string $sourcePath
+	 * @param string $destinationPath
+	 * @return bool
+	 */
+	public function beforeMove(string $sourcePath, string $destinationPath): bool {
+		try {
+			$node = $this->server->tree->getNodeForPath($sourcePath);
+			$destinationNode = $this->server->tree->getNodeForPath(dirname($destinationPath));
+		} catch (\Exception $e) {
+			return true;
+		}
+
+		// Check if a file is being restored before proceeding
+		if (!$node instanceof ITrash || !$destinationNode instanceof RestoreFolder) {
+			return true;
+		}
+
+		$fileInfo = $node->getFileInfo();
+		if (!$fileInfo instanceof ITrashItem) {
+			return true;
+		}
+		$restoreFolder = dirname($fileInfo->getOriginalLocation());
+		$freeSpace = $this->view->free_space($restoreFolder);
+		$filesize = $fileInfo->getSize();
+		if ($freeSpace >= 0 && $freeSpace < $filesize) {
```

Comment:
```suggestion
		if ($freeSpace < $filesize) {
```
This should be redundant

Reply:
There are times when `view->free_space()` returns negative values mapping to `FileInfo::SPACE_NOT_COMPUTED`, `FileInfo::SPACE_UNKNOWN` and `FileInfo::SPACE_UNLIMITED` which is why I had the `$freeSpace >= 0` check to ignore and allow restoring in those cases but I realize it might be better to check against those constants explicitly.

Reply:
Oh I totally forgot about that! Yes I think it makes sense to check for the constants explicitly so that the code is more self-explanatory.